nemo/collections/asr/data/audio_to_text.py

```diff
@@ -1066,13 +1066,14 @@ def __init__(
                 begin_idx = (len(audio_tar_filepaths) // world_size) * global_rank
                 end_idx = begin_idx + (len(audio_tar_filepaths) // world_size)
                 audio_tar_filepaths = audio_tar_filepaths[begin_idx:end_idx]
+                self.length = len(self.collection) // world_size
                 logging.info(
                     "Partitioning tarred dataset: process (%d) taking shards [%d, %d)", global_rank, begin_idx, end_idx
                 )
 
             elif shard_strategy == 'replicate':
+                self.length = len(self.collection) // world_size
```

Comment:
In replication mode, full collection is available to all nodes. So why // world_size?

Reply:
fixed.